cli/internal/core/engine.go

```diff
@@ -156,6 +156,13 @@ func (e *Engine) Prepare(options *EngineBuildingOptions) error {
 	taskNames := options.TaskNames
 	tasksOnly := options.TasksOnly
 
+	// If there are no affected packages, we don't need to go through all this work
+	// we can just exit early.
+	// TODO(mehulkar): but we still need to validate bad task names?
```

Comment:
This TODO in https://github.com/vercel/turbo/pull/3902, so we don't block this PR